src/game/AI/ScriptDevAI/scripts/outland/shattrath_city.cpp

```diff
@@ -699,6 +704,257 @@ bool QuestRewarded_npc_adal(Player* player, Creature* creature, Quest const* que
     return false; // unhandled
 }
 
+enum recruitEventIds
+{
+    NPC_GRAND_ANCHORITE_ALMONEN = 19216,
+    NPC_COMMANDER_STEELE = 25141,
+    NPC_F_DRAENEI_TRAINEE = 25137,
+    NPC_M_DRAENEI_TRAINEE = 25136,
+    NPC_F_BLOODELF_TRAINEE = 25135,
+    NPC_M_BLOODELF_TRAINEE = 25134,
+    NPC_BLOODELF_VETERAN = 25143,
+};
+
+struct npc_shattered_sun_traineeAI : public ScriptedAI
+{
+    npc_shattered_sun_traineeAI(Creature* pCreature) : ScriptedAI(pCreature) { Reset(); }
+
+    const float radian = M_PI_F/2;
+    const float forward = 5.497790;
+
+    void ReceiveAIEvent(AIEventType type, Unit* sender, Unit* invoker, uint32 miscValue) override
+    {
+        if (!sender || (sender->GetEntry() != NPC_COMMANDER_STEELE && sender->GetEntry() != NPC_GRAND_ANCHORITE_ALMONEN))
+            return;
+
+        switch (type)
+        {
+            case AI_EVENT_CUSTOM_EVENTAI_A:
+                if (miscValue == 26)
+                    m_creature->SetStandState(UNIT_STAND_STATE_STAND);
+                else if (miscValue == 68)
+                    m_creature->SetStandState(UNIT_STAND_STATE_KNEEL);
+                else if (miscValue == 253)
+                    m_creature->HandleEmote(urand(0,2) ? 253 : 4);
+                else
+                    m_creature->HandleEmote(miscValue);
+
+            break;
+            case AI_EVENT_CUSTOM_EVENTAI_B:
+            {
+                SetRootSelf(true);
+                switch (miscValue)
+                {
+                    case 0:
+                        m_creature->SetOrientation(forward);
+                        break;
+                    case 1:
+                        m_creature->SetOrientation(forward - radian);
+                        break;
+                    case 2:
+                        m_creature->SetOrientation(forward - 2 * radian);
+                        break;
+                    case 3: 
+                        m_creature->SetOrientation(forward + radian);
+                        break;
+                }
+                m_creature->SendHeartBeat();
+            }
+                break;
+            case AI_EVENT_CUSTOM_EVENTAI_C:
+                // Unused
```

Comment:
`AI_EVENT_CUSTOM_EVENTAI_C` is reserved for use if I manage to move SteeleAI to SpawnGroups